src/nal/sps.rs

```diff
@@ -278,33 +278,48 @@ impl From<ProfileIdc> for u8 {
     }
 }
 
-pub struct ScalingList {
-    // TODO
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum ScalingList<const S: usize> {
+    NotPresent,
+    UseDefault,
+    List(Box<[i32; S]>),
```

Comment:
It looks like the possible bounds of `scalingList[ j ]` are `[1, 256)`; is that right? If so, perhaps a `NonZeroU8` would be appropriate rather than an `i32`?

Reply:
I've had a look at "Scaling list syntax" in the spec, and agree it seems to make sense to use `NonZeroU8` here :+1: